### ISSUE TITLE
VA Forms System Core [Component]: Incorrect usage of aria-labels

### ISSUE THREAD

**Issue by TiffanyPender** [department-of-veterans-affairs/va-forms-system-core#545]
-[508-defect-1: ❗️Critical. Must be fixed before launch](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-508-defect-1)

## User Story or Problem Statement

As a screen reader user, I expect the label and hint text to correctly identify the form field and provide proper guidance. When form fields provide conflicting hint text to users, this could be a point of frustration or confusion, especially for users with cognitive issues.

## Details

All iterations of the MONTH and DAY dropdown components include an unnecessary `aria-label` with incorrect information.

*Examples*
* `<va-select>` component for MONTH lists `aria-label` as “Please enter two digits for the month”. The dropdown list contains a list of text months, not two-digit numbers.
* `<va-select>` component for DAY lists `aria-label` as “Please enter two digits for the day”. The user cannot enter two digits for days 1 through 9.

## Acceptance Criteria

- [ ] MONTH and DAY components have `aria-label` removed
- [ ] *optional* Hint text moved into a separate element and referenced on the `<select>` element with `aria-describedby`

## Environment

* macOS Monterey, `<Version 12.4>`
* Chrome, `<Version 102.0.5005.61>`
* VoiceOver
* Staging

## Steps to Recreate

1. Enter `https://staging.va.gov/burial-poc-v6/` in browser address bar
2. Navigate through all steps that contain dropdowns for MONTH and DAY.
3. Verify that the `aria-label` verbiage is incorrect on the `<va-select>` components for each MONTH and DAY element

## Proposed Solution (if known)

In general, `aria-label` attribute should be used in cases where an element has no accessible name or there is no content in the DOM that can be referenced with the `aria-labelledby` attribute. The MONTH and DAY components already have a `<label>` available. In this particular case, I have a couple of recommendations:
* Completely removing the `aria-label` attribute from the `<va-select>` component
* It seems that the goal of the original text was to give screen reader users a clue to the format of the dropdown so they could begin typing to make a selection. In this case, I recommend updating the text to more clearly state the correct hint, and including it in a separate element that can be referenced with `aria-describedby` on the `<select>` element.

## WCAG or Vendor Guidance (optional)

* [MDN - aria-label](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label)
* [2.4.6 Headings and Labels](https://www.w3.org/WAI/WCAG21/quickref/#headings-and-labels)
* [3.3.2 Labels and Instructions](https://www.w3.org/WAI/WCAG21/quickref/#labels-or-instructions)

**Comment by taylorkaren** [comment]
This could be an issue with the component itself. If it is, this ticket should be transferred to the Design System Team. Do this by:
- Opening the issue in GitHu
- Choose Transfer issue (will be down the page on the right side)
- Choose department-of-veterans-affairs/va.gov-team repo as the repo to transfer the ticket to
- Add a vsp-design-system-team label to the ticket
- Remove the VA Forms System Core label from the ticket.
- Ticket should end up on this board: https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/teams#workspaces/platform-design-system-5f8de67192551b0012ebb802/board
